ros/CMakeLists.txt

```diff
@@ -47,11 +47,13 @@ if("$ENV{ROS_VERSION}" STREQUAL "1")
     COMPONENTS geometry_msgs
                nav_msgs
                sensor_msgs
+               geometry_msgs
                roscpp
                rosbag
                std_msgs
                tf2
-               tf2_ros)
+               tf2_ros
+               Sophus)
```

Comment:
From this one, I'm not quite sure what is the "best" practice... KISS-ICP has its own, isolated and self-contained build system... This means that the only way you couldn't "link" against on of its dependencies is when you copy-paste by hand some source files. 

On the other hand, if we add sophus here, then, we should also add eigen, TBB, etc...  

Do you have a better idea for this?

Reply:
Hmmm, that's a good question. I added the Sophus dependency because I got a linker error for Sophus when doing the following build process for ROS 1:
```
cd ~/catkin_ws/src
git clone https://github.com/PRBonn/kiss-icp
cd ~/catkin_ws
rosdep install --from-paths src --ignore-src -r -y
catkin build
# the compiler initially complained that Sophus could not be found
# but worked once I added the dependency to CMakeLists.txt and package.xml
```

I think that, because we explicitly include Sophus and Eigen headers in the OdometryServer code, we should have a dependency on these in package.xml and CMakeLists.txt. That way, if kiss-icp-cpp ever somehow removed its dependencies on these packages in the future (unlikely) then the ROS packages would still compile. It feels risky to include a dependency's dependency based on prior knowledge of the build process without explicitly adding it to the package.xml.

(I'm keen to know if there's a better way to do this too.)

Reply:
Linking error is unlikely since sophus is a header-only library. On the other hand, lets take for example `tf2_ros`, just because is the one above: 
https://github.com/ros/geometry2/blob/309814fd7d55d874f9bac40b2cc8669a5b86f3c5/tf2_ros/CMakeLists.txt#L9

We don't need to include `actionlib` as our dependency. That's the whole magic of build systems.

On the other other hand, I also agree that just because we explicitly include the headers in the Odometryserver, that injects a hard dependency.... so it might be cleaner to explicitly put it there. My problem is with ROS-based build systems, that's if you put sophus on the `package.xml` and it's not in your system, then the build should fail. In contrast, KISS will fetch this at build time...

I'm puzzled about which is the best approach for this case. I will meditate and be back at this PR